sdk/core/core-lro/samples/v3/javascript/customPoller.js

```diff
@@ -0,0 +1,175 @@
+    },
+    pollUntilDone: async (pollOptions) => {
+      resultPromise ??= (async () => {
+        await statePromise;
+        if (!state) {
+          throw new Error("Poller should be initialized but it is not!");
+        }
+        const { abortSignal: inputAbortSignal } = pollOptions || {};
+        // In the future we can use AbortSignal.any() instead
+        function abortListener() {
+          abortController.abort();
+        }
+        const abortSignal = abortController.signal;
+        if (inputAbortSignal?.aborted) {
+          abortController.abort();
+        } else if (!abortSignal.aborted) {
+          inputAbortSignal?.addEventListener("abort", abortListener, { once: true });
+        }
+
+        try {
+          if (!poller.isDone) {
+            await poller.poll({ abortSignal });
+            while (!poller.isDone) {
+              await delay(currentPollIntervalInMs, { abortSignal });
+              await poller.poll({ abortSignal });
+            }
+          }
+        } finally {
+          inputAbortSignal?.removeEventListener("abort", abortListener);
+        }
+        switch (state.status) {
+          case "succeeded":
+            return poller.result;
+          case "canceled":
+            throw new Error(cancelErrMsg);
+          case "failed":
+            throw state.error;
+          case "notStarted":
+          case "running":
+            throw new Error(`Polling completed without succeeding or failing`);
+        }
+      })().finally(() => {
+        resultPromise = undefined;
+      });
+      return resultPromise;
+    },
+    async poll(pollOptions) {
+      await statePromise;
+      if (!state) {
+        throw new Error("Poller should be initialized but it is not!");
+      }
+      switch (state.status) {
+        case "succeeded":
+          return state;
+        case "canceled":
+          throw new Error(cancelErrMsg);
+        case "failed":
+          throw state.error;
+      }
+      await pollOperation({
+        state,
+        options: pollOptions,
+        setDelay: (pollIntervalInMs) => {
+          currentPollIntervalInMs = pollIntervalInMs;
+        },
+      });
+      await handleProgressEvents();
+      switch (state.status) {
+        case "canceled":
+          throw new Error(cancelErrMsg);
+        case "failed":
+          throw state.error;
+      }
+
+      return state;
+    },
+    then(onfulfilled, onrejected) {
+      return poller.pollUntilDone().then(onfulfilled, onrejected);
+    },
+    catch(onrejected) {
+      return poller.pollUntilDone().catch(onrejected);
+    },
+    finally(onfinally) {
+      return poller.pollUntilDone().finally(onfinally);
+    },
+    [Symbol.toStringTag]: "Poller",
+  };
+  return poller;
+}
+
+async function main() {
+  const poller = createPoller({ intervalInMs: 1000 });
+  const res = await poller.pollUntilDone();
+  console.log(res);
+}
+
+main().catch((err) => {
+  console.error("The sample encountered an error:", err);
```

Comment:
Should we `process.exit(1)` here or something to indicate the sample failed? Otherwise we'll never know (assuming we even set it to run on CI)

Reply:
Good point, addressed in https://github.com/Azure/azure-sdk-for-js/pull/32469/commits/9babca3eabce977777fdbb2aba1959735d48fcd2